projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/DataJob.java

```diff
@@ -54,6 +54,10 @@ public class DataJob {
   @Column(name = "last_execution_duration")
   private Integer lastExecutionDuration;
 
+  @OneToOne(mappedBy = "dataJob", cascade = CascadeType.ALL, orphanRemoval = true)
+  @PrimaryKeyJoinColumn
```

Comment:
Is this foreign key relationshiop needed? 

I believe the data job service calls the deployment service to clean up deployments on delete.

Seems depending only on the Service or API level is better than having dependencies on the database level as well. 

I would like to keep Deployment Service, Jobs Service, Properties Services, Secrest Service , Execution Service almost as separate (micro)services  with the same monolith. Aka Modular Monolith architecture.

Reply:
The current solution does not require an explicit call to the DeploymentService for deletion, since the DataJob entity is the leading one.